owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.kt

```diff
@@ -0,0 +1,451 @@
+                openFile()
+                return true
+            }
+
+            R.id.action_see_details -> {
+                seeDetails()
+                return true
+            }
+
+            R.id.action_send_file -> {
+                requireActivity().sendDownloadedFilesByShareSheet(listOf(file))
+                return true
+            }
+
+            R.id.action_set_available_offline -> {
+                val fileToSetAsAvailableOffline = ArrayList<OCFile>()
+                fileToSetAsAvailableOffline.add(file)
+                fileOperationsViewModel.performOperation(SetFilesAsAvailableOffline(fileToSetAsAvailableOffline))
+                return true
+            }
+
+            R.id.action_unset_available_offline -> {
+                val fileToUnsetAsAvailableOffline = ArrayList<OCFile>()
+                fileToUnsetAsAvailableOffline.add(file)
+                fileOperationsViewModel.performOperation(FileOperation.UnsetFilesAsAvailableOffline(fileToUnsetAsAvailableOffline))
+                return true
+            }
+
+            else -> {
+                return super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mTextLoadTask.apply {
+            cancel(true)
+            dismissLoadingDialog()
+        }
+        isOpen = false
+        currentFilePreviewing = null
+    }
+
+    override fun onFileMetadataChanged(updatedFile: OCFile?) {
+        if (updatedFile != null) {
+            file = updatedFile
+        }
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    override fun onFileMetadataChanged() {
+        mContainerActivity.storageManager?.let {
+            file = it.getFileByPath(file.remotePath)
+        }
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    override fun onFileContentChanged() {
+        loadAndShowTextPreview()
+    }
+
+    override fun updateViewForSyncInProgress() {
+        mProgressController.showProgressBar()
+    }
+
+    override fun updateViewForSyncOff() {
+        mProgressController.hideProgressBar()
+    }
+
+    private fun openFile() {
+        mContainerActivity.fileOperationsHelper.openFile(file)
+        finish()
+    }
+
+    private fun seeDetails() {
+        mContainerActivity.showDetails(file)
+    }
+
+    private fun finish() {
+        activity?.onBackPressed()
+    }
+
+    companion object {
+        private const val EXTRA_FILE = "FILE"
+        private const val EXTRA_ACCOUNT = "ACCOUNT"
+        var isOpen = false
+        var currentFilePreviewing: OCFile? = null
+
+        fun newInstance(file: OCFile, account: Account): PreviewTextFragment {
+            val args = Bundle().apply {
+                putParcelable(EXTRA_FILE, file)
+                putParcelable(EXTRA_ACCOUNT, account)
+            }
+
+            return PreviewTextFragment().apply {
+                arguments = args
+            }
+        }
+
+        @JvmStatic
```

Comment:
What would happen if we remove this label?

Reply:
I removed the `JvmStatic` label because it's only necessary when you are using a `Java` class and you need to call the method that is in a `Kotlin` class